### ISSUE TITLE
"Status line is too long" error message when URL is too long

### ISSUE THREAD

**Issue by mmb** [fafhrd91/aiohttp#7177]
### Describe the bug

The "Status line is too long" error in cb_on_url is confusing because the URL is too long, not the status line. It looks like it might have been copied from cb_on_status.

https://github.com/aio-libs/aiohttp/blob/master/aiohttp/_http_parser.pyx#L671

### To Reproduce

1. Request a URL that exceeds the length limit
2. See the "Status line is too long" message

### Expected behavior

I expect a more accurate error message like "URL is too long".

### Logs/tracebacks

```python-traceback
aiohttp.http_exceptions.LineTooLong: 400, message='Got more than 16380 bytes (27594) when reading Status line is too long.'
```

### Python Version

```console
$ python --version
Python 3.8.0
```

### aiohttp Version

```console
$ python -m pip show aiohttp
```

### multidict Version

```console
$ python -m pip show multidict
```

### yarl Version

```console
$ python -m pip show yarl
```

### OS

Linux

### Related component

Server

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by webknjaz** [comment]
Fair enough. If you end up sending a PR, make sure to add a test verifying that the behavior is the same for Cython and pure-Python implementations.

**Comment by Dreamsorcerer** [comment]
These are basically the same though. I think it should just say request line instead of status line here.

Strictly speaking, we should probably be using on_method and on_version to track the length of the entire request line. It's currently possible to exceed the line length using a long method.

**Comment by Dreamsorcerer** [comment]
I think this message got removed in 3.13.
